### PR TITLE
Upgrade all demos to 2.0

### PR DESCRIPTION
**Pull request by dirkgr**
This can't be merged until https://github.com/allenai/allennlp-models/pull/227 is merged.

Still missing:
 - [ ] Order of questions for RC needs to be changed so that SQuAD questions come first

### CODE REVIEW

File: demo [hunk 1]
```diff
@@ -160,8 +160,8 @@ class Container:
         """
         Builds the container.
         """
-        printFlush(f"Building {self.cid}")
-        bp = subprocess.run(shlex.split(self.build_cmd()), encoding="utf-8", capture_output=True)
+        printLnFlush(f"Building {self.cid}")
+        bp = subprocess.run(shlex.split(self.build_cmd()))
```
Comment (dirkgr) [inline demo:164]: This makes the demo script print the build steps to console.

**Comment by dirkgr** [comment]
Actually, we can change the order of the questions independently of this PR. @jonborchardt, can you give it a review? I just merged the corresponding change in -models.

File: demo (hunk 1 above)
Comment (jonborchardt) [reply demo:164]: 👍

File: ui/src/demos/reading-comprehension/types.ts [hunk 2]
```diff
@@ -163,59 +163,18 @@ export const isNAQANetPredictionArithmetic = (
...
-        typedPred.program_nested_expression !== undefined
-    );
-};
-
 export type Prediction =
     | BiDAFPrediction
     | TransformerQAPrediction
-    | NAQANetPrediction
-    | NMNPrediction;
+    | NAQANetPrediction;
```
Comment (jonborchardt) [inline ui/src/demos/reading-comprehension/types.ts:169]: there is a glob of code written in ui
all of these files should be deleted:
https://github.com/allenai/allennlp-demo/tree/master/ui/src/demos/reading-comprehension/nmn

this makes me so sad, that took me a long time to write!

**Review by jonborchardt — APPROVED** [review]
be sure to also clean up https://github.com/allenai/allennlp-demo/tree/master/ui/src/demos/reading-comprehension/nmn

im so sad this is getting deleted! i JUST wrote it!

File: ui/src/demos/reading-comprehension/types.ts (hunk 2 above)
Comment (dirkgr) [reply ui/src/demos/reading-comprehension/types.ts:169]: 😢

**Comment by dirkgr** [comment]
So since I was already bringing back NMN, I also brought back semantic parsing. Though it still doesn't have a front-end, and I think that's by design.

**Comment by dirkgr** [comment]
@jonborchardt, do you want to review this again?

**Review by jonborchardt — APPROVED** [review]
lg2m, did you test that each demo returns predictions and attacks work?

**Comment by dirkgr** [comment]
I just tested predictions. But @schmmd has said for a long time that he wanted to test the demo :-P

Should I spin up an environment?

**Comment by dirkgr** [comment]
I would [spin it up right now](https://marina.apps.allenai.org/a/allennlp-demo/e/create), but I don't want it running all night while nobody is looking anyways.

**Comment by schmmd** [comment]
@dirkgr why don't you deliver to prod, and then I can take a look there ;-)

**Comment by dirkgr** [comment]
I deployed this to https://allennlp-demo.demomodelsweep.apps.allenai.org/, and it does not work one bit. Glad I tried it first. I'm not sure what's wrong, but I don't have time this week anymore, so I'm putting this aside until next week.

**Comment by dirkgr** [comment]
We believe this is a problem with the ad-hoc, not a problem with this code. So we're shipping it!

**Pull request merged**